syntax = "proto3";

package cheers.message;

message ColorRGB {
  float r = 1;
  float g = 2;
  float b = 3;
}

message ColorRGBA {
  float r = 1;
  float g = 2;
  float b = 3;
  float a = 4;
}

message Vector2i {
  int32 x = 1;
  int32 y = 2;
}

message Vector2f {
  float x = 1;
  float y = 2;
}

message Vector3i {
  int32 x = 1;
  int32 y = 2;
  int32 z = 3;
}

message Vector3f {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Vector4f {
  float x = 1;
  float y = 2;
  float z = 3;
  float w = 4;
}

message Pose {
  Vector4f orientation = 2;
  Vector3f position = 3;
}

message Triangle {
  Vector3i index = 1;
  Vector3f normal = 2;
}
